### PR TITLE
StandaloneMmPkg: Add LockBox Dependency Library

### PR DESCRIPTION
**Pull request by xieyuanh**
# Description

<_Include a description of the change and why this change was made._>

<_For each item, place an "x" in between `[` and `]` if true. Example: `[x]` (you can also check items in GitHub UI)_>

<_Create the PR as a Draft PR if it is only created to run CI checks._>

<_Delete lines in \<\> tags before creating the PR._>

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

<_Describe the test(s) that were run to verify the changes._>

## Integration Instructions

<_Describe how these changes should be integrated. Use N/A if nothing is required._>

### CODE REVIEW

File: StandaloneMmPkg/Library/SmmLockBoxMmDependency/SmmLockBoxMmDependency.inf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  VERSION_STRING                 = 1.0
+  LIBRARY_CLASS                  = NULL
+  CONSTRUCTOR                    = SmmLockBoxMmDependencyConstructor
+
+[Sources]
+  SmmLockBoxMmDependency.c
+
+[Packages]
+  MdePkg/MdePkg.dec
+  StandaloneMmPkg/StandaloneMmPkg.dec
```
Comment (jiaxinwu) [inline StandaloneMmPkg/Library/SmmLockBoxMmDependency/SmmLockBoxMmDependency.inf:27]: one small comment: looks we don't need StandaloneMmPkg.dec?

**Comment by ardbiesheuvel** [comment]
Please explain why this library exists and how it will be used.

Standalone MM is standalone and is not like SMM which overlaps with DXE, and has access to the EFI protocol database etc. Standalone MM does not have such access, and so I don't understand why you are adding a library of type DXE_DRIVER here.

**Review by jiaxinwu — APPROVED** [review]

**Comment by xieyuanh** [comment]
> Please explain why this library exists and how it will be used.
> 
> Standalone MM is standalone and is not like SMM which overlaps with DXE, and has access to the EFI protocol database etc. Standalone MM does not have such access, and so I don't understand why you are adding a library of type DXE_DRIVER here.

Hi Ard, since standalone MM cannot utilize gBS, we make is as a DXE driver. I also explain standaloned MM lockbox related logic. This patch is part of that.

The LockBox Dependency Library is designed for standalone MM environments where gBS are not accessible to indicates that LockBox API is ready for use.

For DXE drivers use lockbox APIs via a communication mechanism triggering an SMI, it's must to have the corresponding SMI handler pre-installed for interrupt management. (There will be Standalone MM Lockbox Driver. whose entry point includes the registration of a communication handler and a notification for the SMM Ready To Lock Protocol, ensuring secure transitions to lock phase.)

To ensure orderly operations and proper notification, besides specified the guid in the [Depex] section of the .inf file. The installation of smi handler, along with the LockBox protocol marked by gEfiLockBoxProtocolGuid, must be informed to the DXE drivers. This protocol installation signifies that the LockBox API is ready for use, and this functionality is implemented in the constructor of this library.

**Review by ardbiesheuvel — APPROVED** [review]

**Review by jiaxinwu — APPROVED** [review]

**Comment by xieyuanh** [comment]
Hi All, I have updated the outdated code base just now, with no code changes. could you please approve it again? If there is no problem could you please merge the code. Thank you very much!

**Review by jiaxinwu — APPROVED** [review]

**Review by niruiyu — APPROVED** [review]

**Pull request merged**